smart_open/tests/test_smart_open.py

```diff
@@ -679,15 +679,15 @@ def test_open_with_keywords(self):
         """This test captures Issue #142."""
         fpath = os.path.join(CURR_DIR, 'test_data/cp852.tsv.txt')
         with open(fpath, 'rb') as fin:
-            expected = fin.read().decode('cp852')
+            expected = fin.read().decode('cp852').replace('\r\n', '\n')
```

Comment:
This is hacky. We shouldn't have to tweak the expected value like this. Remember, smart_open needs to behave exactly the same as the built-in open function in such conditions.

I think either:

1. We need to specify the line terminator explicitly when reading under Windows, or
2. There is a bug with how we handle line terminators under Windows

Do you have time to investigate?

Reply:
Yes, I have researched this question.

1. Cannot specify line terminator when opening files in binary mode.
2. When opened in text mode, Windows automatically replaces \p \n with \n. This conversion does not occur when the string is decoded. So I applied an explicit replacement.

I used this documentation:
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
In text mode, the default when reading is to convert platform-specific line endings (\ n on Unix, \ r \ n on Windows) to just \ n. When writing in text mode, the default is to convert occurrences of \ n back to platform-specific line endings.

If this option is incorrect, I am ready to think more.

Reply:
OK then, so why not open both files in text mode? It seems like the binary flag is irrelevant here.